source/dagger/gameplay/common/aiming_system.cpp

```diff
@@ -0,0 +1,70 @@
+#define _USE_MATH_DEFINES
+
+#include "aiming_system.h"
+
+#include "core/graphics/sprite.h"
+#include "core/input/inputs.h"
+
+#include <cmath>
+#include <math.h>
+
+void AimingSystem::OnInitialize(Registry& registry_, Entity entity_)
+{
+    InputReceiver& receiver = registry_.get<InputReceiver>(entity_);
+    for (auto command : { "rotate" })      //Input context has to have a key bound for "rotate" command (or you can rename the command to something you like)
+    {
+        receiver.values[command] = 0;
+    }
+}
+
+void AimingSystem::Run()
+{
+    Engine::Registry().view<InputReceiver, Sprite, Crosshair>().each(
+        [](const InputReceiver input_, Sprite& sprite_, Crosshair& crosshair_) //The sprite component with the same entity as this crosshair component is of a sprite that is used for center of rotation (like character for example)
+        {
+            Float32 rotate = input_.values.at("rotate");        //Get the input value of the rotate command - the amount the angle changes if the button is pressed 
```

Comment:
add a comment above with "rotate" setup from ini file, so other teams can easily set it up.